packages/api-i18n/src/graphql/context.ts

```diff
@@ -219,6 +219,26 @@ const createBaseContextPlugin = () => {
             __i18n.locales = await loadLocales();
         };
 
+        const withEachLocale: I18NContextObject["withEachLocale"] = async (locales, cb) => {
+            const initialLocale = getDefaultLocale();
+            if (!initialLocale) {
+                return;
+            }
+            const results = [];
+            for (const locale of locales) {
+                setContentLocale(locale);
+                setCurrentLocale("default", locale);
+                try {
+                    results.push(await cb(locale));
+                } finally {
+                    setContentLocale(initialLocale);
+                    setCurrentLocale("default", initialLocale);
+                }
+            }
+
+            return results;
+        };
+
```

Comment:
Once this is merged, I would create a separate PR, to make this work with AsyncLocalStorage, so that this `withEachLocale` supports parallel execution. I'll be working on the same thing for tenants, so it would be nice to have this all in 5.40.